bindings/profilers/wall.cc

```diff
@@ -1089,15 +1088,14 @@ void WallProfiler::OnGCStart(v8::Isolate* isolate) {
   if (curCount == 0) {
     gcAsyncId = GetAsyncIdNoGC(isolate);
   }
-  gcCount.store(curCount + 1, std::memory_order_relaxed);
   std::atomic_signal_fence(std::memory_order_release);
+  gcCount.store(curCount + 1, std::memory_order_relaxed);
 }
 
 void WallProfiler::OnGCEnd() {
   auto newCount = gcCount.load(std::memory_order_relaxed) - 1;
-  std::atomic_signal_fence(std::memory_order_acquire);
+  std::atomic_signal_fence(std::memory_order_acq_rel);
   gcCount.store(newCount, std::memory_order_relaxed);
```

Comment:
This makes me wonder what prevents compiler from reordering `gAsyncId = -1` before `gcCount.store(newCount, std::memory_order_relaxed);` ?
In the previous code, I believe `std::atomic_signal_fence(std::memory_order_release);` was preventing it.
It questions again my whole understanding of `atomic_signal_fence`...

Reply:
In this case at least the effect is benign, as it's a cleanup – we could even omit setting it to -1 and just leave the stale value in. This indeed doesn't answer your larger point.

Reply:
You might as well remove the fence, since it's not doing anything here.